crates/core/src/crud_vtab.rs

```diff
@@ -69,15 +76,22 @@ extern "C" fn disconnect(vtab: *mut sqlite::vtab) -> c_int {
     ResultCode::OK as c_int
 }
 
-
 fn begin_impl(tab: &mut VirtualTable) -> Result<(), SQLiteError> {
     let db = tab.db;
 
-    let insert_statement = db.prepare_v3("INSERT INTO ps_crud(tx_id, data) VALUES (?1, ?2)", 0)?;
+    const SQL: &str = formatcp!("\
+WITH insertion (tx_id, data) AS (VALUES (?1, ?2))
```

Comment:
To check if there is data in the patch, it might be more efficient to just check `data->'data' != '{}'` - `json_each` could potentially add some overhead here. I haven't done any actual benchmarks though - the overhead may be negligible.